spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/RabbitListenerAnnotationBeanPostProcessorTests.java

```diff
@@ -273,6 +278,42 @@ public void queuesToDeclare() {
 		context.close();
 	}
 
+	@Test
+	public void concurrency() throws InterruptedException, ExecutionException {
+		final int concurrencyLevel = 8;
+		final ExecutorService executorService = Executors.newFixedThreadPool(concurrencyLevel);
+		try {
+			for (int i = 0; i < 1000; ++i) {
```

Comment:
The test was initially created to reproduce the problem I had noticed in logs. That;s the only purpose of having it. With the new `applicationContext` on each iteration there are *1000* times that contention happens, so there are more chances to reproduce. 

With the new implementation that problem can't happen, so theoretically test is not needed and can be ignored, as @garyrussell suggested